app/routes/reference/$id.tsx

```diff
@@ -0,0 +1,96 @@
+import { Form, useLoaderData } from "@remix-run/react"
+import {
+  type ActionArgs,
+  redirect,
+  type LoaderArgs,
+} from "@remix-run/server-runtime"
+import { useState } from "react"
+import invariant from "tiny-invariant"
+import {
+  type det_references,
+  getReferenceById,
+  updateReferenceById,
+} from "~/models/reference.server"
+import {
+  getNumberFromFormInput,
+  getStringFromFormInput,
+} from "~/utils/generalUtils"
+
+export async function loader({ params }: LoaderArgs) {
+  invariant(params.id)
+  const id: number = parseInt(params.id)
+  const data: det_references | null = await getReferenceById(id)
+  if (!data) throw new Error(`Reference with id: ${id} could not be obtained`)
+
+  return data
+}
+
+export async function action({ request, params }: ActionArgs) {
+  const data = Object.fromEntries(await request.formData())
+  invariant(data.id)
+  invariant(data.shortDisplay)
+  invariant(data.referenceText)
+
+  const id = getNumberFromFormInput(data.id)
+  const shortDisplay = getStringFromFormInput(data.shortDisplay)
+  const referenceText = getStringFromFormInput(data.referenceText)
+
+  console.log(id)
```

Comment:
No need to fix this now, but in the future let's clean up any `console.log`s. Yes I am guilty of this myself. Haha.

Reply:
LOL yup